src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -188,6 +188,35 @@ public function getBoolean($key, $default = false)
     }
 
     /**
+     * Returns the parameter value converted to a DateTime object.
+     *
+     * @param string             $key      The parameter key
+     * @param string             $format   The expected date format
+     * @param string             $default  The default value to be converted to a DateTime object if the parameter key does not exist
```

Comment:
Though, the other getter methods accept `mixed` argument which works if you only do a simple type cast (for ints, for example, as an int cast to int is still an int). So I think we should still allow (but not require) to pass a `\DateTime` object here and simply not transform it when it is used.